WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift

```diff
@@ -46,4 +46,30 @@ public final class SingleProductScreen: ScreenObject {
         XCTAssertTrue(app.staticTexts["TIP"].exists)
         XCTAssertTrue(app.textViews[productName].exists)
     }
+
+    public func verifyProductTypeScreenLoaded(productType: String) throws -> Self {
+        let priceLabel = NSPredicate(format: "label CONTAINS[c] %@", "price")
+        let inventoryLabel = NSPredicate(format: "label CONTAINS[c] %@", "inventory")
+        let productTypeLabel = NSPredicate(format: "label CONTAINS[c] %@", productType)
+        let addVariationLabel = NSPredicate(format: "label CONTAINS[c] %@", "add variations")
+
+        // the common fields on add product screen
+        XCTAssertTrue(app.cells["product-review-cell"].exists)
+        XCTAssertTrue(app.staticTexts.containing(productTypeLabel).firstMatch.exists)
+
+        // different product types displays different fields on add product screen
+        // this is to validate that the correct screens are displayed
+        switch productType {
+        case "physical", "virtual":
+            XCTAssertTrue(app.staticTexts.containing(priceLabel).firstMatch.exists)
+            XCTAssertTrue(app.staticTexts.containing(inventoryLabel).firstMatch.exists)
+        case "variable":
+            XCTAssertTrue(app.staticTexts.containing(addVariationLabel).firstMatch.exists)
+            XCTAssertTrue(app.staticTexts.containing(inventoryLabel).firstMatch.exists)
+        default:
+            // fail test if product type doesn't exist
```

Comment:
I think the line below this comment has a good descriptive error message, so it's not necessary:

```suggestion
```

Reply:
removed in https://github.com/woocommerce/woocommerce-ios/pull/8407/commits/c9a64bce98ca8c9a44005cc30b3e2e8e5a22e02b